File: C/lt_01_estseq/lt01_estseq_33.c
#include <stdio.h>

void serie(double *sum, int *bottom) {
  if (*bottom > 0) {
    *sum += (1.0 / *bottom);
    *bottom -= 1;
    serie(sum, bottom);
  }
}

int main() {
  double sum = 0;
  int num = 0;
  printf("Insira o valor de N: ");
  scanf("%d", &num);

  serie(&sum, &num);

  printf("O valor da serie sera %.2lf.\n", sum);
  return 0;
}

File: rust/lt01-estseq/lt01-estseq-33.rs
use std::io;

fn main() {
    let num = input_f64("Insira um número: ");
    println!("O valor da série com {:.0} termos é {:.2}", num, serie(num));
}

fn serie(mut num: f64) -> f64 {
    let mut serie = 0.0;
    while num > 0.0 {
        serie += 1.0 / num;
        num -= 1.0;
    }
    return serie;
}

fn input_f64(texto: &str) -> f64 {
    let int: f64;
    loop {
        println!("{}", texto);
        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Error read_line");
        int = match input.trim().parse() {
            Ok(n) => {
                if n >= 0.0 {
                    n
                } else {
                    println!("Valor inválido!");
                    continue;
                }
            }
            Err(_) => {
                println!("Valor inválido!");
                continue;
            }
        };
        return int;
    }
}
